src/software/thunderscope/gl/layers/gl_referee_info_layer.py

```diff
@@ -0,0 +1,249 @@
+
+                self.placement_tolerance_graphic.set_position(
+                    self.ball_placement_point.x_meters,
+                    self.ball_placement_point.y_meters,
+                )
+                self.placement_tolerance_graphic.show()
+
+                self.placement_target_graphic.set_position(
+                    self.ball_placement_point.x_meters,
+                    self.ball_placement_point.y_meters,
+                )
+                self.placement_target_graphic.set_radius(
+                    BALL_PLACEMENT_TOLERANCE_RADIUS_METERS
+                )
+                self.placement_target_graphic.show()
+
+                self.robot_avoid_circle_graphic.set_position(
+                    self.ball_placement_point.x_meters,
+                    self.ball_placement_point.y_meters,
+                )
+                self.robot_avoid_circle_graphic.show()
+
+                self.ball_placement_countdown_graphic.setData(
+                    text=f"{BALL_PLACEMENT_TIME_LIMIT_S}s",
+                    pos=[
+                        self.ball_placement_point.x_meters
+                        + BALL_PLACEMENT_ROBOT_AVOID_RADIUS_METERS / 2,
+                        self.ball_placement_point.y_meters
+                        + BALL_PLACEMENT_ROBOT_AVOID_RADIUS_METERS
+                        + 0.1,
+                        0,
+                    ],
+                )
+                self.ball_placement_countdown_graphic.show()
+
+                self.placement_start_time = time.time()
+                self.ball_placement_point_hidden = False
+
+            # shrinking or expanding placement target graphic
+            if self.shrink_target:
+                self.placement_target_graphic.set_radius(
+                    self.placement_target_graphic.radius - 0.01
+                )
+                self.shrink_target = self.placement_target_graphic.radius >= 0
+            else:
+                self.placement_target_graphic.set_radius(
+                    self.placement_target_graphic.radius + 0.01
+                )
+                self.shrink_target = (
+                    self.placement_target_graphic.radius
+                    >= BALL_PLACEMENT_TOLERANCE_RADIUS_METERS
+                )
+
+            # update the count-down graphics
+            time_left = max(
+                int(self.cached_referee_info["currentActionTimeRemaining"]) // 1000000,
+                0,
+            )
+            self.ball_placement_countdown_graphic.setData(text=f"{time_left}s")
+
+    def __update_referee_info(self):
+        """Update gamestate and command info text displays"""
+        referee_proto = self.referee_vis_buffer.get(block=False, return_cached=False)
+        if not referee_proto:
+            return
+
+        referee_msg_dict = MessageToDict(referee_proto)
+        if not referee_msg_dict:
+            return
+        self.cached_referee_info = referee_msg_dict
+
+        if not self.gamestate_type_text:
```

Comment:
nit: these 2 if statements are pretty identical, could probably pull them out to a new method

Reply:
Done

Reply:
you can probably parametrize a class-level member with a `lambda` but that might be overkill since there's only 2